src/coreclr/CMakeLists.txt

```
@@ -141,14 +141,10 @@ endif(NOT CLR_CMAKE_TARGET_ARCH_WASM)
 
 
 if(NOT CLR_CROSS_COMPONENTS_BUILD)
-<<<<<<< HEAD
-    add_subdirectory(nativeaot)
-=======
   # NativeAOT only buildable for a subset of CoreCLR-supported configurations
-  if((CLR_CMAKE_HOST_LINUX OR CLR_CMAKE_HOST_OSX OR CLR_CMAKE_HOST_WIN32) AND (CLR_CMAKE_HOST_ARCH_ARM64 OR CLR_CMAKE_HOST_ARCH_AMD64) AND NOT (CLR_CMAKE_HOST_OSX AND CLR_CMAKE_HOST_ARCH_ARM64))
+  if(((CLR_CMAKE_HOST_LINUX OR CLR_CMAKE_HOST_OSX OR CLR_CMAKE_HOST_WIN32) AND (CLR_CMAKE_HOST_ARCH_ARM64 OR CLR_CMAKE_HOST_ARCH_AMD64 OR CLR_CMAKE_TARGET_ARCH_WASM) AND NOT (CLR_CMAKE_HOST_OSX AND CLR_CMAKE_HOST_ARCH_ARM64)) OR (EMSCRIPTEN AND CLR_CMAKE_TARGET_ARCH_WASM))
```

```suggestion
  if(((CLR_CMAKE_HOST_LINUX OR CLR_CMAKE_HOST_OSX OR CLR_CMAKE_HOST_WIN32 OR EMSCRIPTEN) AND (CLR_CMAKE_HOST_ARCH_ARM64 OR CLR_CMAKE_HOST_ARCH_AMD64 OR CLR_CMAKE_TARGET_ARCH_WASM) AND NOT (CLR_CMAKE_HOST_OSX AND CLR_CMAKE_HOST_ARCH_ARM64)))
```
If I understand correctly the conditions here, this seems clearer.

Alternatively, we could `elseif (CLR_CMAKE_TARGET_ARCH_WASM) /* add_subdirectory(nativeaot) */`. It has a lower probability of conflicts arising (e. g. when OSX ARM64 becomes supported).